QEfficient/transformers/models/gptj/modeling_gptj.py

```diff
@@ -0,0 +1,408 @@
+# -----------------------------------------------------------------------------
+#
+# Copyright (c)  2023-2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# -----------------------------------------------------------------------------
+
+from typing import Optional, Tuple, Union
+
+import torch
+from torch import nn
+from torch.nn import CrossEntropyLoss
+from transformers.cache_utils import DynamicCache
+from transformers.modeling_outputs import (
+    BaseModelOutputWithPast,
+    CausalLMOutputWithPast,
+)
+from transformers.models.gptj.modeling_gptj import(
+    get_embed_positions,
+    GPTJAttention,
+    GPTJModel,
+    GPTJForCausalLM,
+    logger,
+    rotate_every_two,
+)
+from transformers.utils.import_utils import is_torch_fx_proxy
+
+from QEfficient.transformers.modeling_attn_mask_utils import _create_causal_mask
+
+def apply_rotary_pos_emb(tensor: torch.Tensor, sin: torch.Tensor, cos: torch.Tensor) -> torch.Tensor:
```

Comment:
Can this be imported from `transformers.models.gptj.modeling_gptj `too?

Reply:
This is required to avoid torch.repeat_interleave function getting called in HF code, need to replace that with reshapes and just repeat.

Reply:
i guess direct should work? do you see any issues if this is not modified?

Reply:
![image](https://github.com/quic/efficient-transformers/assets/168400541/154bb39d-fbb9-4173-9384-660e77f3cb04)
these unsupported are coming due to torch.interleave